src/pocketmine/Player.php

```diff
@@ -2864,6 +2864,10 @@ public function handleDropItem(DropItemPacket $packet) : bool{
 			// Windows 10 Edition drops the contents of the crafting grid on container close - including air.
 			return true;
 		}
+		
+		if($this->gamemode === self::SPECTATOR) {
```

Comment:
I never said it was noticeable, I prefer $this->gamemode === self::SPECTATOR over $this->isSpectator(), if you prefer it the other way you're welcome to close the PR and someone else can make a new one.